src/interchange/src/protobuf.rs

```diff
@@ -104,68 +96,30 @@ impl DecodedDescriptors {
 pub struct Decoder {
     descriptors: DecodedDescriptors,
     packer: Row,
-    ccsr_client: Option<ccsr::Client>,
-    validated_schema_id: Option<i32>,
+    confluent_wire_format: bool,
 }
 
 impl Decoder {
     /// Constructs a decoder for a particular Protobuf message.
     pub fn new(
         descriptors: DecodedDescriptors,
-        schema_registry: Option<ccsr::ClientConfig>,
+        confluent_wire_format: bool,
     ) -> Result<Self, anyhow::Error> {
         Ok(Decoder {
             descriptors,
             packer: Default::default(),
-            ccsr_client: schema_registry.map(|sr| sr.build()).transpose()?,
-            validated_schema_id: Default::default(),
+            confluent_wire_format,
         })
     }
 
     /// Decodes the encoded Protobuf message into a [`Row`].
     pub async fn decode(&mut self, mut bytes: &[u8]) -> Result<Option<Row>, anyhow::Error> {
-        if let Some(client) = &self.ccsr_client {
-            let (schema_id, adjusted_bytes) = crate::confluent::extract_protobuf_header(bytes)?;
-
-            if let Some(validated_schema_id) = self.validated_schema_id {
-                if validated_schema_id != schema_id {
-                    bail!(
-                        "cannot decode protobuf, expected schema id: {}, found {}; schema evolution in protobuf is not supported. \
-                        See https://github.com/MaterializeInc/materialize/issues/9598 for more details.",
-                        validated_schema_id,
-                        schema_id
-                    );
-                }
-            } else {
-                let compiled = compile_proto(schema_id, client).await?;
-                let schema_to_compare = DecodedDescriptors::from_bytes(
-                    &strconv::parse_bytes(&compiled.schema)?,
-                    // Needs to match the name exactly
-                    self.descriptors.message_name.clone(),
-                )?;
-                if schema_to_compare.message_descriptor.get_proto()
-                    == self.descriptors.message_descriptor.get_proto()
-                {
-                    self.validated_schema_id = Some(schema_id);
-                } else {
-                    bail!(
-                        "cannot decode protobuf, schema id: {} refers to a schema different than the expected schema; \
-                        schema evolution in protobuf is not supported. \
-                        See https://github.com/MaterializeInc/materialize/issues/9598 for more details.",
-                        schema_id
-                    );
-                }
-            }
+        if self.confluent_wire_format {
+            let (_schema_id, adjusted_bytes) = crate::confluent::extract_protobuf_header(bytes)?;
```

Comment:
nit: add a comment here explaining why we ignore the schema_id?

Reply:
Done.